src/citrine/resources/file_link.py

```diff
@@ -289,3 +297,30 @@ def _complete_upload(self, dest_name: str, uploader: _Uploader):
 
         url = self._get_path(file_id) + '/versions/{}'.format(version)
         return FileLink(filename=dest_name, url=url)
+
+    def download(self, file_link: FileLink, local_path: str):
+        """
+        Download the file associated with a given FileLink to the local computer.
+
+        Parameters
+        ----------
+        file_link: FileLink
+            Resource referencing the external file.
+        local_path: str
+            Path to save file on the local computer. If `local_path` is a directory,
+            then the filename of this FileLink object will be appended to the path.
+
+        """
+        directory, filename = os.path.split(local_path)
+        if not filename:
+            filename = file_link.filename
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
```

Comment:
I think you want `os.mkdirs` in case there are multiple levels of directory given, like `my/really/annoying/path`

Reply:
Good catch